lib/diagnostic.js

```diff
@@ -5,10 +5,13 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (array, callback) {
+let  minimum =  array.reduce(min);
```

Comment:
```min()``` is the function that we are writing to pass into ```reduce()```.  We don't want to call ```reduce()``` on ```min()``` here.  Think of this function as a callback function that we're writing for somebody else to use when they invoke ```reduce()``` on an array.